mesop/runtime/context.py

```diff
@@ -120,9 +125,36 @@ def serialize_state(self) -> pb.States:
       states.states.append(pb.State(data=serialize_dataclass(state)))
     return states
 
+  def diff_state(self) -> pb.States:
+    states = pb.States()
+    for (key, state), previous_state in zip(
+      self._states.items(), self._previous_states.values()
+    ):
+      states.states.append(pb.State(data=diff_state(previous_state, state)))
+      # If state has not changed no need to update the previous state since it should be
+      # the same.
+      #
+      # One thing to note is that there seems to be an issue multiple states with the
+      # same key can exist (i.e. usage in Mesop demo app). It's unclear if this causes
```

Comment:
This seems like a weird bug. Can you file an issue?

Reply:
Bug: https://github.com/google/mesop/issues/391